decidim-proposals/app/controllers/decidim/proposals/admin/valuation_assignments_controller.rb

```diff
@@ -12,7 +12,7 @@ def create
           Admin::AssignProposalsToValuator.call(@form) do
             on(:ok) do |_proposal|
               flash[:notice] = I18n.t("valuation_assignments.create.success", scope: "decidim.proposals.admin")
-              redirect_to EngineRouter.admin_proxy(current_component).root_path
+              redirect_to evaluate_from_single_proposal_url || evaluate_from_proposals_list_url
```

Comment:
```suggestion
              redirect_to after_add_evaluator_url
```
I would use a single method here, something like:

```ruby
def after_add_evaluator_url
  return request.referer if request.referer.present? && request.referer.include?(:proposal_id.to_s)

 request.referer
end
```